### PR TITLE
Fix other locations where we create an invalid codeql-zip-archive uri

### PR DESCRIPTION
**Pull request by aeisenberg**
Also, create a convenience function for generating a codeql-zip-archive
at the root of the archive.

## Checklist

- [n/a] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [n/a] Issues have been created for any UI or other user-facing changes made by this pull request.
- [n/a] `@github/docs-content-dsp` has been cc'd in all issues for UI or other user-facing changes made by this pull request.

### CODE REVIEW

File: extensions/ql-vscode/src/databases.ts [hunk 1]
```diff
@@ -439,10 +436,7 @@ export class DatabaseItemImpl implements DatabaseItem {
     const sourceArchive = this.sourceArchive;
     if (sourceArchive === undefined || !sourceArchive.fsPath.endsWith('.zip'))
       return undefined;
-    return encodeSourceArchiveUri({
-      pathWithinSourceArchive: '/',
```
Comment (adityasharad) [inline extensions/ql-vscode/src/databases.ts:443]: Does the slash matter? It will now be empty.

**Review by adityasharad — APPROVED** [review]
Looks sensible, minor questions. Is it feasible to have integration tests in place for the contextual providers that would have caught this?

**Comment by adityasharad** [comment]
Oh genuine Windows failure - I think a test needs path normalisation.

**Comment by aeisenberg** [comment]
> Looks sensible, minor questions. Is it feasible to have integration tests in place for the contextual providers that would have caught this?

Sorry. Missed this comment. I will add some integration tests.

**Pull request merged**